Index: vendor/plugins/calendar_date_select/lib/calendar_date_select.rb
===================================================================
--- vendor/plugins/calendar_date_select/lib/calendar_date_select.rb	(revision 226)
+++ vendor/plugins/calendar_date_select/lib/calendar_date_select.rb	(working copy)
@@ -138,7 +138,7 @@
         out << " "
         
         out << image_tag(CalendarDateSelect.image, 
-            :onclick => "new CalendarDateSelect( $(this).previous(), #{options_for_javascript(calendar_options)} );",
+            :onclick => "new CalendarDateSelect( $(this).previous().hasClassName('fieldWithErrors') ? $(this).previous().down() : $(this).previous(), #{options_for_javascript(calendar_options)} );", 
             :style => 'border:0px; cursor:pointer;')
       end
       
